From: =?utf-8?q?Fran=C3=A7ois_Cartegnie?= <fcvlcdev@free.fr>
Date: Tue, 24 Sep 2024 18:43:08 +0700
Subject: codec: x265: handle X265_MAX_FRAME_THREADS on newer CPU

libx265 will bail out when hitting the limit

x265 [error]: frameNumThreads (--frame-threads) must be [0 .. X265_MAX_FRAME_THREADS)

(cherry picked from commit bbd294d1e46b86653873ec88d49e9a15cbaae68f)
---
 modules/codec/x265.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/codec/x265.c b/modules/codec/x265.c
index 5585f2c..2943ea6 100644
--- a/modules/codec/x265.c
+++ b/modules/codec/x265.c
@@ -36,6 +36,10 @@
 
 #include <x265.h>
 
+#ifndef X265_MAX_FRAME_THREADS
+# define X265_MAX_FRAME_THREADS 16
+#endif
+
 /*****************************************************************************
  * Module descriptor
  *****************************************************************************/
@@ -156,6 +160,8 @@ static int  Open (vlc_object_t *p_this)
     x265_param_default(param);
 
     param->frameNumThreads = vlc_GetCPUCount();
+    if(param->frameNumThreads > X265_MAX_FRAME_THREADS)
+        param->frameNumThreads = X265_MAX_FRAME_THREADS;
     param->bEnableWavefront = 0; // buggy in x265, use frame threading for now
     param->maxCUSize = 16; /* use smaller macroblock */
 
